pkg/plugins/resources/k8s/store.go

```diff
@@ -59,6 +62,14 @@ func (s *KubernetesStore) Create(ctx context.Context, r core_model.Resource, fs
 	}
 
 	labels, annotations := SplitLabelsAndAnnotations(opts.Labels, obj.GetAnnotations())
+	if labels == nil {
+		labels = map[string]string{}
+	}
+	if core_model.IsLocallyOriginated(s.CpMode, r) {
```

Comment:
On Kubernetes, the resource can be created in 2 ways:
* applied by user, we catch that in webhook `defaulter.go`
* created by CP, we catch that in this `store.go`

Should we share a call to `ComputeLabels` in these two places?